journal/src/main/java/io/camunda/zeebe/journal/file/SegmentWriter.java

```diff
@@ -230,26 +229,24 @@ private void reset(final long index, final long lastFlushedIndex) {
     } catch (final BufferUnderflowException e) {
       // Reached end of the segment
     } catch (final CorruptedJournalException e) {
-      handleChecksumMismatch(e, nextIndex, lastFlushedIndex, position);
+      if (detectCorruptionAsPartialWrite) {
+        resetPartiallyWrittenEntry(e, position);
+      } else {
+        throw e;
+      }
```

Comment:
:upside_down_face: 
```suggestion
      if (!detectCorruptionAsPartialWrite) {
        throw e;
      }
      
      resetPartiallyWrittenEntry(e, position);
```

Reply:
Though if we do it as a guard clause, then we may want to avoid the negation and just make the param `detectCorruption` or `strictCorruptionHandling` or something. Although truth be told, I'd rather avoid boolean arguments in general, but it also feels too small to refactor into a strategy thing :smile: